WordPress/gutenberg
# [Mobile] - List block - Fix merging nested list into paragraph

**Related PRs:**
- https://github.com/WordPress/gutenberg/pull/50634
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5785



## What?
This PR fixes an issue when merging a List Item that has sub-times into a Paragraph removes its nested blocks.

## Why?
To fix an issue with the List block.

## How?
By bringing the [fix made for the web editor](https://github.com/WordPress/gutenberg/pull/50634) to the mobile editor as well. It also includes a fix for the undo functionality related to this block as well.

## Testing Instructions

**Precondition**: Use the following HTML:

```html

<p>A quick brown fox.</p>


<ul>
<li>One
<ul>
<li>Two</li>


<li>Three</li>
</ul>
</li>
</ul>

```

- Open the editor
- Set the above snippet
- Tap on the first item of the list at the beginning of the TextInput
- Tap on the backspace button
- **Expect** that item to be converted into a Paragraph block and its sub-items to remain as a list.
- Tap undo and **expect** the list to be shown as it was initially.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 

Before | After
-|-
<video src="https://github.com/WordPress/gutenberg/assets/4885740/7a7491ad-b888-4627-bdbf-332559eedcec" width=200 /> | <video src="https://github.com/WordPress/gutenberg/assets/4885740/cf43f00a-b6d0-496c-bb73-3f0668563d4a" width=200 />
